tests/backend_test.py

```diff
@@ -1374,3 +1374,21 @@ def test_unitary_sim_gateset() -> None:
     unitary_sim_gateset = backend.backend_info.gate_set
     unsupported_ops = {OpType.Reset, OpType.Measure, OpType.Conditional}
     assert unitary_sim_gateset.isdisjoint(unsupported_ops)
+
+
+def test_unitary_backend_transpiles() -> None:
+    """regression test for https://github.com/CQCL/pytket-qiskit/issues/142"""
+    backend = AerUnitaryBackend()
+    n_ancillas = 5  # using n_ancillas <=4 doees not raise an error
+    n_spins = 1
+    circ = Circuit(n_ancillas + n_spins)
+    trgt = Circuit(n_spins)
+    trgt.X(0)
+
+    circ.add_qcontrolbox(
+        QControlBox(CircBox(trgt), n_ancillas), list(range(n_ancillas + n_spins))
+    )
+
+    compiled_circ = backend.get_compiled_circuit(circ, optimisation_level=0)
+    # using optimisation_level >= 1 does not raise an error
+    backend.run_circuit(compiled_circ)
```

Comment:
Would be nice to have some sort of assertion in the test to check we got the expected result.

Reply:
I agree, do you have a suggestion? Otherwise I'll comment that we are testing that run_circuit doesn't throw

Reply:
Something like:
```python
r = backend.run_circuit(compiled_circ)
u = r.get_unitary()
assert np.isclose(u[62:64, 62:64], np.asarray([[0., 1.].[1., 0.]]))
```
I think (may have got the syntax wrong, please check). I.e. check that the lower-right 2x2 submatrix of the unitary is the matrix of the X gate.